docs/source/1.0/spec/aws/aws-core.rst

```diff
@@ -1219,6 +1219,301 @@ Clients SHOULD use an LRU cache implementation with an initial cache limit of
+
+Client behavior
+===============
+
+HTTP request checksums
+----------------------
+
+When a client calls an operation which has the ``httpChecksum`` trait where
+``requestChecksumRequired``is set to ``true``, the client MUST include a
+checksum in the HTTP request.
+
+When a client calls an operation which has the ``httpChecksum`` trait where
+``requestAlgorithmMember`` is set, the client MUST look up the input member
+represented by ``requestAlgorithmMember`` property. The value of this member is
+the checksum algorithm that the client MUST use to compute the request payload
+checksum.
+
+The computed checksum MUST be supplied at a resolved location as per the
+:ref:`resolving checksum location <aws.protocols#httpChecksum-trait_resolving-checksum-location>`
+section. If the resolved location is ``header``, the client MUST put the
+checksum into the HTTP request headers. If the resolved location is
+``trailer``, the client MUST put the checksum into the `chunked trailer part`_
+of the body. The header or trailer name to use with an algorithm is resolved as
+per the :ref:`resolving checksum name <aws.protocols#httpChecksum-trait_resolving-checksum-name>`
+section.
+
+If no ``requestAlgorithmMember`` is set, the client MUST compute an MD5
+checksum of the request payload and place it in the ``Content-MD5`` header.
+
+If the HTTP header corresponding to a checksum is set explicitly, the client
+MUST use the explicitly set header and skip computing the payload checksum.
+
+.. seealso:: See :ref:`client behavior<aws.protocols#httpChecksum-trait_header-conflict-behavior>`
+    for more details.
+
+HTTP response checksums
+-----------------------
+
+When a client receives a response for an operation which has the ``httpChecksum``
+trait where the ``requestValidationModeMember`` property is set, the client
+MUST look up the input member represented by the property's value. If the input
+member is set to ``ENABLED``, the client MUST perform best-effort validation of
+checksum values returned in the HTTP response.
+
+A client MUST use the list of supported algorithms modeled in the
+``responseAlgorithms`` property to look up the checksum(s) for which validation
+MUST be performed. The client MUST look for the HTTP header in the response as
+per the :ref:`resolving checksum name <aws.protocols#httpChecksum-trait_resolving-checksum-name>`
+section.
+
+A client MUST raise an error if the response checksum to validate does not
+match computed checksum of the response payload for the same checksum algorithm.
+
+If a checksum is provided in a response that is not listed in the
+``responseAlgorithms`` property, the client MUST ignore the value and MUST NOT
+attempt to validate it.
+
+A client MUST provide a mechanism for customers to identify whether checksum
+validation was performed on a response and which checksum algorithm was
+validated.
+
+.. _chunked trailer part: https://tools.ietf.org/html/rfc7230#section-4.1.2
```

Comment:
Is this the right link? I thought S3 had a custom version of this that isn't actually a trailer.